util/src/main/java/io/camunda/zeebe/util/FeatureFlags.java

```diff
@@ -22,19 +22,36 @@ public record FeatureFlags(/*boolean foo*/ ) {
    *   As we gain more confidence in the efficacy of the feature flag, the
    *   default value can be set to 'true'
    *
+   * - define a default value to be used in tests
+   *
+   * - make sure that all relevant tests use the default value for tests
+   *
    * - add a field, getter and setter to FeatureFlagsCfg
    *
    * - add a description of the feature flag to
    *    - dist/src/main/config/broker.standalone.yaml.template
    *    - dist/src/main/config/broker.yaml.template
    *
+   * - add test cases to FeaturesFlagCfgTest and feature-flags-cfg.yaml
+   *
    * Be careful with parameter order in constructor calls!
    */
 
   //  protected static final boolean FOO_DEFAULT = false;
-  //
+
+  private static final boolean YIELDING_DUE_DATE_CHECKER = false;
```

Comment:
:thought_balloon: Why do we not enable the yielding by default? 

If enabled then it avoids that the partition is unhealthy. For me, it sounds more important that the partition is healthy than the timers are far behind.

If the feature is disabled then we don't get feedback if it is causing troubles.

Reply:
@npepinpe This is one of the feature flags we want to test in dev/int and bring into production as soon as possible.

I agree with Philipp that setting it to on is probably the lesser of two evils. I wanted to err on the side of caution, though. 
Also, my manual tests revealed that this change alone does not mitigate the issue that @Zelldon complained about.

I still think it's a step in the right direction, though.